tests/guest-tools/unix/test_guest_tools_unix.py

```diff
@@ -0,0 +1,124 @@
+        detected_distro = vm.distro()
+        assert detected_distro == vm_distro
+
+    def test_install(self, running_vm, state):
+        vm = running_vm
+
+        # skip test for windows and some unixes
+        if vm.is_windows:
+            pytest.skip('Test module only valid for Unix VMs')
+        state.vm_distro = vm.distro()
+        if state.vm_distro == "alpine":
+            pytest.skip('Alpine not supported by the guest tools installation script at the moment')
+
+        # Check that we are able to detect that xe-daemon is running
+        assert vm.ssh_with_result(['pgrep', '-x', 'xe-daemon']).returncode == 0, \
+            "xe-daemon must be running and detected by pgrep"
+
+        # remove the installed tools
+        print("Detect package manager and uninstall the guest tools")
+        pkg_mgr = vm.detect_package_manager()
+        if pkg_mgr == 'rpm':
+            # Our guest tools come in two packages for RPM distros: xe-guest-utilities and xe-guest-utilities-xenstore.
+            # However, the following implementation will also work for a single xe-guest-utilities RPM.
+            vm.execute_script('rpm -qa | grep xe-guest-utilities | xargs rpm -e')
+        elif pkg_mgr == 'apt-get':
+            vm.ssh(['apt-get', 'remove', '-y', 'xe-guest-utilities'])
+        else:
+            pytest.skip("Package manager '%s' not supported in this test" % pkg_mgr)
+
+        # check that xe-daemon is not running anymore
+        assert vm.ssh_with_result(['pgrep', '-x', 'xe-daemon']).returncode != 0, \
+            "xe-daemon must not be running anymore"
+
+        # mount ISO
+        print("Mount guest tools ISO")
+        vm.mount_guest_tools_iso()
+        tmp_mnt = vm.ssh(['mktemp', '-d'])
+        vm.ssh(['mount', '/dev/cdrom', tmp_mnt])
+
+        # get tools version number for future checks
+        prefix = 'xe-guest-utilities_'
+        suffix = '_x86_64.tgz'
+        tgz_filename = vm.ssh(['find', tmp_mnt, '-name', prefix + '*' + suffix])
+        state.tools_version = tgz_filename.split('/')[-1][len(prefix):-len(suffix)]
+
+        # install tools
+        print("Install tools %s using install.sh" % state.tools_version)
+        vm.ssh([tmp_mnt + '/Linux/install.sh', '-n'])
+
+        # unmount ISO
+        print("Unmount guest tools ISO")
+        vm.ssh(['umount', tmp_mnt])
+        vm.unmount_guest_tools_iso()
+
+        # check that xe-daemon is running
+        wait_for(lambda: vm.ssh_with_result(['pgrep', '-x', 'xe-daemon']).returncode == 0,
+                 "Wait for xe-daemon running")
+
+    def test_check_tools(self, running_vm, state):
+        vm = running_vm
+        self._check_tools_version(vm, state.tools_version)
+        self._check_os_info(vm, state.vm_distro)
+
+    def test_check_tools_after_reboot(self, running_vm, state):
+        vm = running_vm
+        vm.reboot(verify=True)
+        self._check_tools_version(vm, state.tools_version)
+        self._check_os_info(vm, state.vm_distro)
```

Comment:
This tests are not independant, if 1st fails, 2nd makes no sense.

Reply:
Yes, the tests in this series depend on each other. I didn't want to put the whole installation test into a fixture because 
, that's why the class is decorated with `@pytest.mark.incremental`. If a test fails, the others are automatically skipped.